src/api/procedures/__tests__/initiateCorporateAction.ts

```diff
@@ -0,0 +1,337 @@
+    corporateActionKindToCaKindStub = sinon.stub(
+      utilsConversionModule,
+      'corporateActionKindToCaKind'
+    );
+    dateToMomentStub = sinon.stub(utilsConversionModule, 'dateToMoment');
+    checkpointToRecordDateSpecStub = sinon.stub(
+      utilsConversionModule,
+      'checkpointToRecordDateSpec'
+    );
+    stringToTextStub = sinon.stub(utilsConversionModule, 'stringToText');
+    targetsToTargetIdentitiesStub = sinon.stub(utilsConversionModule, 'targetsToTargetIdentities');
+    percentageToPermillStub = sinon.stub(utilsConversionModule, 'percentageToPermill');
+    stringToIdentityIdStub = sinon.stub(utilsConversionModule, 'stringToIdentityId');
+  });
+
+  beforeEach(() => {
+    addTransactionStub = procedureMockUtils.getAddTransactionStub().returns([rawCaId]);
+    initiateCorporateActionTransaction = dsMockUtils.createTxStub(
+      'corporateAction',
+      'initiateCorporateAction'
+    );
+
+    maxDetailsLengthQueryStub = dsMockUtils.createQueryStub('corporateAction', 'maxDetailsLength', {
+      returnValue: dsMockUtils.createMockU32(100),
+    });
+    mockContext = dsMockUtils.getContextInstance();
+
+    stringToTickerStub.withArgs(ticker, mockContext).returns(rawTicker);
+    corporateActionKindToCaKindStub.withArgs(kind, mockContext).returns(rawKind);
+    dateToMomentStub.withArgs(declarationDate, mockContext).returns(rawDeclDate);
+    checkpointToRecordDateSpecStub.withArgs(checkpoint, mockContext).returns(rawRecordDate);
+    stringToTextStub.withArgs(description, mockContext).returns(rawDetails);
+    targetsToTargetIdentitiesStub.withArgs(targets, mockContext).returns(rawTargets);
+    percentageToPermillStub.withArgs(defaultTaxWithholding, mockContext).returns(rawTax);
+    percentageToPermillStub
+      .withArgs(taxWithholdings[0].percentage, mockContext)
+      .returns(rawWithholdings[0][1]);
+    stringToIdentityIdStub.withArgs('someDid', mockContext).returns(rawWithholdings[0][0]);
+  });
+
+  afterEach(() => {
+    entityMockUtils.reset();
+    procedureMockUtils.reset();
+    dsMockUtils.reset();
+  });
+
+  afterAll(() => {
+    entityMockUtils.cleanup();
+    procedureMockUtils.cleanup();
+    dsMockUtils.cleanup();
+  });
+
+  test('should throw an error if there are more targets than the maximum allowed', async () => {
+    const proc = procedureMockUtils.getInstance<Params, CAId>(mockContext);
+
+    let err;
+
+    try {
+      await prepareInitiateCorporateAction.call(proc, {
+        ticker,
+        kind,
+        declarationDate,
+        checkpoint,
+        description,
+        targets: {
+          identities: range(1001).map(() => 'someDid'),
+          treatment: TargetTreatment.Include,
+        },
+        defaultTaxWithholding,
+        taxWithholdings,
+      });
+    } catch (error) {
+      err = error;
+    }
+
+    expect(err.message).toBe('Too many target Identities');
+    expect(err.data).toEqual({
+      maxTargets: 1000,
```

Comment:
This will fail when you start fetching this value from the chain

Reply:
We'll have to change the test anyway to mock the value being returned